### PR TITLE
make CI work for 1.4

### PR DESCRIPTION
**Pull request by jcantrill**
@richm  this is the code im having issue with the tests passing in CI.  any help you could add would be appreciated.

### CODE REVIEW

**Comment by richm** [comment]
What errors do you get when you run this?

File: hack/testing/logging.sh [hunk 1]
```diff
@@ -7,7 +7,9 @@
 # This scripts starts the OpenShift server with a default configuration.
 # The OpenShift Docker registry and router are installed.
 # It will run the logging tests
-
+if [ ! -z ${DEBUG} ] ; then
```
Comment (richm) [inline hack/testing/logging.sh]:
```
if [ -n "${DEBUG:-}" ] ; then
```

File: hack/testing/logging.sh [hunk 2]
```diff
@@ -112,37 +117,55 @@ trap "cleanup" EXIT
...
 export ARTIFACT_DIR=${ARTIFACT_DIR:-${TMPDIR:-/tmp}/origin-aggregated-logging/artifacts}
 os::util::environment::use_sudo
-os::util::environment::setup_all_server_vars "origin-aggregated-logging/"
+#os::util::environment::setup_all_server_vars "origin-aggregated-logging/"
 
 os::log::system::start
 
+OCP_VERSION="v1.4.1"
 export KUBELET_HOST=$(hostname)
+export CLUSTER_UP_DIR=/tmp/origin-aggregated-logging/cluster
```
Comment (richm) [inline hack/testing/logging.sh]: does /tmp/origin-aggregated-logging/cluster exist at this point?

File: hack/testing/logging.sh [hunk 3]
```diff
@@ -112,37 +117,55 @@ trap "cleanup" EXIT
...
+export KUBECONFIG="${ADMIN_KUBECONFIG}"
 
-os::start::configure_server
+os::cmd::expect_success "docker pull openshift/origin:${OCP_VERSION}"
+
+CLUSTER_UP="oc cluster up --version=${OCP_VERSION} --host-data-dir=${ETCD_DATA_DIR} --host-config-dir=${SERVER_CONFIG_DIR}"
+
+#os::start::configure_server
 if [ -n "${KIBANA_HOST:-}" ] ; then
+    os::cmd::expect_success "$CLUSTER_UP"
```
Comment (richm) [inline hack/testing/logging.sh]: This runs `oc cluster up --version=${OCP_VERSION} ...`

File: hack/testing/logging.sh [hunk 4]
```diff
@@ -112,37 +117,55 @@ trap "cleanup" EXIT
...
+    oc cluster up --version=v1.4.1 --host-data-dir=${ETCD_DATA_DIR} --host-config-dir=${SERVER_CONFIG_DIR}
```
Comment (richm) [inline hack/testing/logging.sh]: This runs the same oc cluster up again?

**Comment by jcantrill** [comment]
[test]

File: hack/testing/logging.sh (hunk 2 above)
Comment (jcantrill) [inline hack/testing/logging.sh]: Are you suggesting maybe I need a mkdir statement? It works on my AMI, but that could just be a fluke

File: hack/testing/logging.sh (hunk 3 above)
Comment (jcantrill) [inline hack/testing/logging.sh]: Yes

File: hack/testing/logging.sh (hunk 4 above)
Comment (jcantrill) [inline hack/testing/logging.sh]: Should be removed... never made it through this code path

File: hack/testing/logging.sh (hunk 2 above)
Comment (richm) [inline hack/testing/logging.sh]: I just didn't see where this is created - but if it is created then don't worry about it.

**Comment by richm** [comment]
@jcantrill the error is coming from here: https://github.com/jcantrill/origin-aggregated-logging/blob/2d93f6d9b9c9ee047de0615b6803667bd2a2ad9b/hack/testing/logging.sh#L303

```
++ oc get dc -o name -l component=es-ops
No resources found.
+ ops_dc=
+ os::cmd::expect_success 'oc patch         -p '\''{"spec":{"template":{"spec":{"containers":[{"name":"elasticsearch","securityContext":{"privileged": true}}]}}}}'\'''
```

The es-ops dc is not found.

**Comment by jcantrill** [comment]
[test]

File: hack/testing/logging.sh [hunk 5]
```diff
@@ -162,7 +173,7 @@ fi
 if [ $NOSETUP = 1 ] ; then
```
Comment (richm) [inline hack/testing/logging.sh]: change this to
```
if [ "$NOSETUP" = 1 ] ; then
```

Comment (richm) [inline hack/testing/logging.sh]: because of this:
```
./logging.sh: line 145: [: =: unary operator expected
```

**Comment by jcantrill** [comment]
@richm @stevekuznetsov Do we see any value in pursuing this?  I ideally I would like for 1.4 to work with our CI but I have concern there is a lot of effort here to make this work with minimal gain.

**Comment by stevekuznetsov** [comment]
1.4 doesn't use the Ansible playbooks for setup, right? Kind of the point of the stuff I was trying to do was to make the tests _only_ care about the things they are testing, not the set-up or the cluster or whatever. So if I succeeded in that, we should be able to slap those tests into this branch and, barring functionality differences, they should be fine. Are you seeing that other work is necessary, too?

**Comment by jcantrill** [comment]
@stevekuznetsov I have not additionally evaluated but it was slow go on my last attempts.  1.4 is not ansible based, correct.

**Comment by stevekuznetsov** [comment]
Yeah, it's hard to say. Supporting 1.4 and 1.5 was definitely in the design space for the changes. I think it would make more sense to focus right now on #443. Once that's finished, move the tests from `master` verbatim back to the old branches and then make edits to the test logic as necessary.

**Comment by jcantrill** [comment]
Closing as this seems like a pipe dream... and less important for now

**Pull request closed without merging**